examples/functions-with-included-record-parameters/functions_with_included_record_parameters.bal

```diff
@@ -0,0 +1,83 @@
+import ballerina/io;
+ 
+type Student record {|
+    string firstName;
+    string lastName?;
+|};
+
+type AdmissionDate record {|
+    string year = "2020";
+    string month = "Jan";
+    never admissionNo?;
+|};
+
+type Grades record {|
+    never math?;
+    never physics?;
+    int...;
+|};
+
+// Adding `*` before the type of the parameter marks the parameter as an included record parameter.
+// The names of the fields in the record type of an included record parameter must be distinct 
+// from each other and also from the names of the other parameters, unless it is an optional field of type `never`. 
+// If not, it results in a compilation error.
+function printStudentDetails(int admissionNo, *Student student) {
+    string name = student.firstName;
+ 
+    string? lastName = student?.lastName;
+    if lastName is string {
+        name += string ` ${lastName}`;
+    }
+ 
+    io:println("Admission No: ", admissionNo, ", Student Name: ", name);
+}  
+
+function printAdmissionDate(int admissionNo, *AdmissionDate date) {
+    string admissionDate = date.year + "-" + date.month;
+
+    io:println("Admission No: ", admissionNo,
+               ", Admission Date: ", admissionDate);
+}
+
+function printAverage(int math, int physics, *Grades grades) {
+    int totalMarks = math + physics;
+    int count = 2;
+ 
+    foreach int grade in grades {
+        totalMarks += grade;
+        count += 1;
+    }
+ 
+    io:println("Average Marks: ", totalMarks/count);
+}
+ 
+public function main() {
+
+    Student student = {
+        firstName: "Anna",
+        lastName: "Christina"
+    };
+    
+    // Call the function by passing a positional argument for the `student` included record parameter.
+    printStudentDetails(1001, student);
+
+    // Call the function by passing values for the fields of the included record parameter. 
+    // Values need to be supplied for the required fields that do not have default values.
+    printStudentDetails(1002, firstName = "Anne", lastName = "Doe");
+
+    // Call the function by passing values only for the required fields of the included record parameter.
+    // The `lastName` field of the included record parameter defaults to `()`. 
```

Comment:
We can't quite say this defaults right? Shall we say
```suggestion
    // Since the `lastName` field is an optional field in the type of the included record parameter
    // it is not mandatory to pass an argument for `lastName`. 
```

Reply:
Fixed.